internal/plugins/manifests/init.go

```diff
@@ -19,18 +19,23 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
+	"path/filepath"
+	"strings"
 
 	"sigs.k8s.io/kubebuilder/v3/pkg/config"
 	cfgv3 "sigs.k8s.io/kubebuilder/v3/pkg/config/v3"
 
 	"github.com/operator-framework/operator-sdk/internal/util/projutil"
 )
 
+// Version of `opm` to download and use for building index images.
+// This version's release artifacts *must* contain a binary for multiple arches; certain releases do not.
+const opmVersion = "v1.15.1"
+
 // RunInit modifies the project scaffolded by kubebuilder's Init plugin.
```

Comment:
Currently, we are adding any default change made by the tool in the manifests plugin. What means manifest? IMO we should decouple this manifest plugin and have one plugin to address its own need such as we have `scorecard.sdk.operatorframework.io/v2` that scaffold the scorecard manifests only. 

This shows that we might be hurting a few concepts with the "manifest" plugin such as single resposanbility and cushion which will not allow us to do actions such as, for example: 

- Tomorrow we might realize that go requires the opm plugin by default but to ansible/helm and then it will not be easily addressed. 
- We can enable/disable the default scaffold to integrate the project with olm/opm 
- We might allow people to use/run/execute only the opm plugin to scaffold only what is required for it to address a use case: 

> As a developer I want to use SDK CLI to integrate my project which was built using KB with OPM so that, I can use the catalog features. 

WDYT about just encapsulate these changes in any opm plugin instead? Do you see any reason for it does not be possible? 

/hold

Reply:
`manifests.sdk.operatorframework.io/v2` already exists, and will be able to be turned off by plugins phase 1.5. Right now there is no option to disable this plugin, but if you wanted to then why would you use operator-sdk over kubebuilder in the first place? This change doesn't break existing `manifests/v2` plugin users, so a new major version/plugin is unnecessary.

Reply:
Hi @estroz, 

> manifests.sdk.operatorframework.io/v2 already exists, 
It is true. We cannot change manifests now. This shipped is sealed, I agree. 👍 
However, why we are doing new RFEs on manifest instead of having their own plugins as we did for the scorecard requirements for example? 

>  This change doesn't break existing manifests/v2 plugin users, so a new major version/plugin is unnecessary.

It is not about break things but follows good practices and do not hurt concepts such as single responsibility which will keep better maintainability for what we are writing now does not become a shoot in our foot in the future.  e.g I can create a method doAllPossibleThings and it will work. Am I right? However, should I do that?

Reply:
The single responsibility of the manifests plugin is to scaffold stuff related to bundles/catalogs. If that isn't clear then perhaps some (go)docs are necessary. Adding features to a plugin does not break SRP if the feature adheres to the responsibility. I would argue that creating a separate plugin for opm creates a maintenance headache because it depends on other things created by the manifests plugin like variables.

In any case, you and I are arguing about semantics while the resulting scaffolded code will look exactly the same regardless of plugin compartmentalization (minus the PROJECT file `plugins` object). Therefore I don't see a point in discussing this, especially considering we can always break out code into separate plugins later without causing breaking change.

Reply:
Oh. I think I got your point. You are saying that makes no sense we have a plugin only for opm without the changes that we do in the scaffolds for olm which are in the manifests plugin.

It is true and fair enough to make sense 👍

Reply:
/hold cancel